### PR TITLE
TRUNK-6202 Replace Hibernate Criteria API with JPA for HibernateFormDAO

### PR DESCRIPTION
**Pull request by k4pran**
## Description of what I changed

This task is to migrate the deprecated hibernate criteria to jpa criteria for the HibernateFormDAO class.

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-6202

## Checklist: I completed these to help reviewers :)

- [X] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [X] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [X] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [X] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [X] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java [hunk 1]
```diff
@@ -43,4 +45,18 @@ public void shouldFilterAgainstFormFields() {
...
 	}
+
+	@Test
+	public void shouldGetFormFieldsByForm() {
+		Form form = new Form(2); 
+		List<FormField> formFields = dao.getFormFields(form);
+
+		assertNotNull(formFields);
+
+		int expectedSize = 2;
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java:56]: Shouldn't the above be final?

Comment (k4pran) [reply api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java]: Updated to make it final. I would rarely make local variables final, do you prefer to do this by default for all variables that are intended to be immutable or are there specific scenarios you want them to be final?

Comment (dkayiwa) [reply api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java]: Is there any reason why you rarely make local variables final?

Comment (k4pran) [reply api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java]: I guess the main reason is other teams I have been on have only used them at class and field level, not so much method and local levels. My preference has been to omit it is that I find it more verbose / less readable with finals. I can see the reasons behind doing it though as it prevents it being changed unintentionally. Regardless, happy to fall in line with the openmrs standard practice

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java [hunk 2]
```diff
@@ -43,4 +45,18 @@ public void shouldFilterAgainstFormFields() {
...
 	}
+
+	@Test
+	public void shouldGetFormFieldsByForm() {
+		Form form = new Form(2); 
+		List<FormField> formFields = dao.getFormFields(form);
+
+		assertNotNull(formFields);
+
+		final int expectedSize = 2;
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java:56]: And our naming convention for constants is using capital letters. 😊

Comment (k4pran) [reply api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java]: done

**Pull request merged**